### PR TITLE
Manual migration

### PR DESCRIPTION
**Pull request by skyfox93**
### To run a migration manually -
import it and list it in the migrations array in the DataSourceConnectionService
call `npm run typeorm:migration-run` to run migrations

### required for transition
Because the existing schema is not reflected in the migrations table, the following ONE TIME step is nessesary to migrate the database to using manual migrations:
call `npm run typeorm:schema-drop` to clear the current database
call `npm run typeorm:migration-run` to make the database and migration table reflect the existing migrations

### required everytime your database is out of date with main
 `npm run typeorm:migration-run` to bring it up to date

### required every time you make changes to an entity on a feature branch
make sure you are migrated up to the existing main:  call `npm run typeorm:migration-run`
call `npm run typeorm:migration-generate` to generate a migration based on the changes you've made
call `npm run typeorm:migration-run` to run the migration

### required everytime before moving back to main after running a migration on a feature branch
`npm run typeorm:migration-revert` to migrate back "down" to main ( if you've added multiple migrations, you must revert multiple times)

### required when merging a pull request into that has migrations
- make sure there were no conflicting or duplicate migrations entered since the pull request was made
- if there were, or if you wish to be safe before merge:

    - on the main branch, make sure your database is up to date and you've , run all pending migrations, then run `npm run typeorom:migration-generate`. It should fail, as you have no made any changes to main.  If migrations are generated when no changes have been made to main, it means that either a) theres a problem with main, or main is missing a migration or b) you made migrations on another branch and forgot to revert them

   - checkout the feature branch, discard any migration files in the pull request and run `npm run typeorm:migration-generate` to re-generate them against the existing main
   - `npm run typeorm:migration-run` and test the pull request works
   - `npm run typeorm:migration-revert` before moving back into main

the following package scripts have been added: 
```
 "typeorm:migration-run": "npm run typeorm migration:run -- -d ./src/datasource.ts",
        "typeorm:migration-generate": gener./migrations/$npm_config_name",
        "typeorm:migration-create": "npm run typeorm -- migration:create ./migrations/$npm_config_name",
        "typeorm:migration-revert": "npm run typeorm -- -d ./src/datasource.ts migration:revert",
        "typeorm:schema-drop": "npm run typeorm -- -d ./src/datasource.ts schema:drop",
        "typeorm:schema-sync": "npm run typeorm -- -d ./src/datasource.ts schema:sync",
```

### CODE REVIEW

**Review by esteban-gs — CHANGES_REQUESTED** [review]
@skyfox93 thanks for putting this together. It's going to work really well with our seed script. One small change would be to update the `README.md` with your notes in the PR description and another small change in the comments.

File: server/package.json [hunk 1]
```diff
@@ -22,6 +22,13 @@
...
         "checks": "npm run && npm run test && npm run lint",
         "prepare": "cd .. && husky install && cd server",
+        "typeorm": "ts-node ./node_modules/typeorm/cli",
+        "typeorm:migration-run": "npm run typeorm migration:run -- -d ./src/datasource.ts",
+        "typeorm:migration-generate": "npm run typeorm -- -d ./src/datasource.ts migration:generate ./migrations/$npm_config_name",
+        "typeorm:migration-create": "npm run typeorm -- migration:create ./migrations/$npm_config_name",
+        "typeorm:migration-revert": "npm run typeorm -- -d ./src/datasource.ts migration:revert",
+        "typeorm:schema-drop": "npm run typeorm -- -d ./src/datasource.ts schema:drop",
+        "typeorm:schema-sync": "npm run typeorm -- -d ./src/datasource.ts schema:sync",
         "seed:drop": "ts-node -r tsconfig-paths/register src/database/seeding/seed.ts"
```
Comment (esteban-gs) [inline server/package.json:32]: While we're looking at scripts, could you rename the `seed:drop` command to another name similar to what you named the other commands and also update README.md

**Comment by skyfox93** [comment]
Updated the readme with a migrations section and renamed `seed:drop` to `db:seed`. (Seeding does not use the typeorm cli so `typeorm:seed` is inappropriate)

**Review by esteban-gs — APPROVED** [review]

**Comment by esteban-gs** [comment]
@skyfox93 , we should be good to merge once we update with latest main version.

**Pull request merged**